src/platform_impl/linux/mod.rs

```diff
@@ -504,20 +510,31 @@ pub enum EventLoop<T: 'static> {
     X(x11::EventLoop<T>),
 }
 
-pub enum EventLoopProxy<T: 'static> {
-    X(x11::EventLoopProxy<T>),
-    Wayland(wayland::EventLoopProxy<T>),
+/// A handle that can be sent across threads and used to wake up the `EventLoop`.
+pub struct EventLoopProxy<T: 'static> {
+    user_sender: calloop::channel::Sender<T>,
 }
```

Comment:
cc @murarth @goddessfreya @vberger for review. What do you guys think of this change?